From c1eaa569a1ebe3aa688ca251fcbcb42ee086cf27 Mon Sep 17 00:00:00 2001
From: dseefeld <dseefeld@microsoft.com>
Date: Wed, 18 Aug 2021 12:20:24 +0000
Subject: [PATCH] Conditionally build allconfigurations

When building portable, only a subset of runtime needs to be
built.  allconfigurations is only needed in a non-portable build.
---
 eng/SourceBuild.props | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eng/SourceBuild.props b/eng/SourceBuild.props
index 7dfc80d41fd..1e3bd504931 100644
--- a/eng/SourceBuild.props
+++ b/eng/SourceBuild.props
@@ -37,7 +37,7 @@
       <InnerBuildArgs>$(InnerBuildArgs) --arch $(TargetRidPlatform)</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) --configuration $(Configuration)</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) --ci</InnerBuildArgs>
-      <InnerBuildArgs>$(InnerBuildArgs) --allconfigurations</InnerBuildArgs>
+      <InnerBuildArgs  Condition="'$(SourceBuildNonPortable)' == 'true'">$(InnerBuildArgs) --allconfigurations</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) --verbosity $(LogVerbosity)</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) --nodereuse false</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) --warnAsError false</InnerBuildArgs>
-- 
2.31.1

